frontend/common/src/components/Pagination/usePagination.ts

```diff
@@ -0,0 +1,117 @@
+import { useMemo, useState } from "react";
+
+export const DOTS = "...";
+
+const range = (start: number, end: number) => {
+  const length = end - start + 1;
+  /*
+  	Create an array of certain length and set the elements within it from
+    start value to end value.
+  */
+  return Array.from({ length }, (_, idx) => idx + start);
+};
+
+/**
+ * A pagination hook that returns the range of numbers to be displayed in out Pagination component as an array.
+ * @param totalCount
```

Comment:
I know we mention this is issue, but when code is informed by outside sources, thoughts on linking it in comment docs for future travallers?

https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/see.html#examples

```suggestion
 * A pagination hook that returns the range of numbers to be displayed in out Pagination component as an array.
 * @see https://www.freecodecamp.org/news/build-a-custom-pagination-component-in-react/
 * @param totalCount
```